client.js

```diff
@@ -56,14 +56,15 @@ var PoolClient = module.exports = function(config, logger){
                             break;
                         case "mining.set_difficulty":
                             _this.difficulty = messageJson.params[0];
-                            _this.target = global.diff1Target.mul(256).div(Math.ceil(_this.difficulty * 256)).toBuffer();
+                            let str = global.diff1Target.multipliedBy(256).div(Math.ceil(_this.difficulty * 256)).toString(16);
+                            _this.target = Buffer.from(str.length % 2 === 0 ? str : "0"+str, 'hex');
```

Comment:
the insertion of a leading 0 if an odd length hex string is found is required, otherwise buffer will parse the str from left to write and everything will be shifted by 4 bits and that's no good